src/main/java/com/limechain/trie/MemoryTrieAccessor.java

```diff
@@ -0,0 +1,167 @@
+package com.limechain.trie;
+
+import com.limechain.runtime.version.StateVersion;
+import com.limechain.storage.DeleteByPrefixResult;
+import com.limechain.storage.trie.TrieStorage;
+import com.limechain.trie.cache.node.PendingInsertUpdate;
+import com.limechain.trie.structure.NodeHandle;
+import com.limechain.trie.structure.TrieNodeIndex;
+import com.limechain.trie.structure.TrieStructure;
+import com.limechain.trie.structure.database.NodeData;
+import com.limechain.trie.structure.nibble.Nibble;
+import com.limechain.trie.structure.nibble.Nibbles;
+import com.limechain.utils.HashUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public abstract sealed class MemoryTrieAccessor extends TrieAccessor
+    permits BlockTrieAccessor, MemoryChildTrieAccessor {
+
+    private final TrieStructure<NodeData> initialTrie;
+    private List<TrieNodeIndex> updates;
+    private DiskTrieService diskTrieService;
```

Comment:
why not final?

Reply:
This is not meant for the MemoryTrieAccessor. It's leftover from testing that I've removed locally after this commit.